plugins/faustwp/includes/replacement/callbacks.php

```diff
@@ -97,94 +97,6 @@ function image_source_srcset_replacement( $sources ) {
 	return $sources;
 }
 
-add_filter( 'preview_post_link', __NAMESPACE__ . '\\post_preview_link', 1000, 2 );
-/**
- * Callback for WordPress 'preview_post_link' filter.
- *
- * Swap the post preview link for headless front-end and to use the API entry to support Next.js preview mode.
- *
- * @param string  $link URL used for the post preview.
- * @param WP_Post $post Post object.
- *
- * @return string URL used for the post preview.
- */
-function post_preview_link( $link, $post ) {
-	$frontend_uri = faustwp_get_setting( 'frontend_uri' );
-
-	if ( $frontend_uri ) {
-		$home_url     = trailingslashit( get_home_url() );
-		$frontend_uri = trailingslashit( $frontend_uri );
-
-		$link = str_replace( $home_url, $frontend_uri, $link );
-
-		// Replace the schemes, if different.
-		$frontend_uri_scheme = wp_parse_url( $frontend_uri, PHP_URL_SCHEME );
-		$link_scheme         = wp_parse_url( $link, PHP_URL_SCHEME );
-		if ( $frontend_uri_scheme !== $link_scheme ) {
-			$link = str_replace( $link_scheme . '://', $frontend_uri_scheme . '://', $link );
-		}
-
-		/**
-		 * This should already be handled by WPE\FaustWP\Replacement\post_link, but
-		 * it's here for verbosity's sake and if the other filter changes for any reason.
-		 */
-		$parsed_link_query = wp_parse_url( $link, PHP_URL_QUERY );
-		$args              = wp_parse_args( $parsed_link_query );
-		$preview_id        = isset( $args['preview_id'] ) ? $args['preview_id'] : $post->ID;
-
-		// Remove ?p=xx&preview=true from link temporarily.
-		$link = remove_query_arg(
-			array_keys( $args ),
-			$link
-		);
-
-		if ( ! isset( $args['previewPathname'] ) ) {
-			$args['previewPathname'] = rawurlencode( wp_make_link_relative( get_permalink( $post ) ) );
-		}
-
-		// Add p=xx if it's missing, which is the case for published posts.
-		if ( ! isset( $args['p'] ) ) {
-			$args['p'] = $preview_id;
-		}
-
-		// Add preview=true in case other plugins have stripped it.
-		if ( ! isset( $args['preview'] ) ) {
-			$args['preview'] = 'true';
-		}
-
-		// Add page_id=xx if it's missing, which is the case for published pages.
-		if ( ! isset( $args['page_id'] ) && 'page' === $post->post_type ) {
-			$args['page_id'] = $preview_id;
-		}
-
-		/**
-		 * We use this query param to get the correct preview post type.
-		 * This saves us a second request to the GraphQL endpoint, as we first
-		 * need to determine the preview node post type before retrieving
-		 * the data.
-		 */
-		$post_type_object = get_post_type_object( $post->post_type );
-		if (
-			! isset( $args['typeName'] ) &&
-			isset( $post_type_object ) &&
-			! empty( $post_type_object->graphql_single_name )
-		) {
-			$gql_type_name    = ucfirst( $post_type_object->graphql_single_name );
-			$args['typeName'] = $gql_type_name;
-		}
-
-		// Add ?p=xx&preview=true to link again.
-		$link = add_query_arg(
-			array(
-				$args,
-			),
-			$link
-		);
-	}
-
-	return $link;
-}
```

Comment:
I think we will still need this code (in a separate function) that we can call to generate the headless preview link.